scipy/optimize/optimize.py

```diff
@@ -475,6 +475,36 @@ def function_wrapper(x, *wrapper_args):
     return ncalls, function_wrapper
 
 
+class _MaxFuncCallError(RuntimeError):
+    pass
+
+
+def _wrap_scalar_function_with_validation(function, args, maxfun):
```

Comment:
This function would need to be tested since we might want to use it elsewhere. I am not certain about having it here and not in some utils.